### PR TITLE
Closes #2518 - Fix back button in PairFragment

### LINKED ISSUE #2518: [Bug] Android back button doesn't work properly in the Pairing QR View
## Steps to reproduce

Turn On Sync -> Pair ~~> Open Camera~~ > Android back button

### Expected behavior
Should go back to Turn On Sync

### Actual behavior
Stays on PairFragment and just closes the camera view.

Turn On Sync -> Pair ~~> Open Camera~~ > Top Left Corner button works properly.

### Device information
* Android device: Essential PH-1 Android 9
* Fenix version: `master` on d894ea6702030cd4fe23f1afbfb0e94661fd9a91

### PR DESCRIPTION
**Pull request by vladikoff**
@ekager r?

What should we do here? Is there a better solution?

### CODE REVIEW

File: app/src/main/java/org/mozilla/fenix/settings/PairFragment.kt [hunk 1]
```diff
@@ -61,10 +61,9 @@ class PairFragment : Fragment(), BackHandler {
     }
 
     override fun onBackPressed(): Boolean {
-        return when {
-            qrFeature.onBackPressed() -> true
-            else -> false
-        }
+        qrFeature.onBackPressed()
+        fragmentManager?.popBackStack()
```
Comment (ekager) [inline app/src/main/java/org/mozilla/fenix/settings/PairFragment.kt]: Generally we shouldn't be using the FragmentManager with the Navigation Library so let's use 
```
findNavController(this@PairFragment).navigateUp()
```

**Comment by vladikoff** [comment]
@ekager I had to use `findNavController(view!!).navigateUp()` How's that?

**Comment by ekager** [comment]
Did you import this `findNavController`?
```
import androidx.navigation.fragment.NavHostFragment.findNavController
```

**Comment by vladikoff** [comment]
@ekager ah I saw it, but didn't use it. Updated now!

**Comment by vladikoff** [comment]
👍

**Comment by vladikoff** [comment]
Thank you!

**Pull request merged**